### ISSUE TITLE
prepare_multisig overwrote used wallet 

### ISSUE THREAD

**Issue by aaronovz1** [monero-project/monero#5586]
Due to a misunderstanding of how wallet RPC works I managed to turn a simple/regular wallet into a multisig wallet. I had two apps simultaneously connected to the same Wallet RPC instance. One app controlled a simple wallet, the other app was working with multisig wallets. In the process, one app called **open_wallet** on the simple wallet, and the other app must have called **prepare_multisig** which to my surprise succeeded in overwriting the simple wallet and turning it into a 2/3 multisig wallet.

I would have expected prepare_multisig/make_multisig to have failed, because the simple wallet had balances and transactions, so should not have been classified as unused.

Is there any way to revert the wallet back to a simple wallet? Am I right in that I would need to separate Wallet RPC instances running so there are no concurrency issues like this?

**Comment by moneromooo-monero** [comment]
prepare_multisig does not change anything. I assume make_multisig was called. I could add a check for "has it been used before", and fail the call, but generally RPC is more "do what I say", rather than "ask the user for confirmation" like monero-wallet-cli.

You can restore your wallet from seed or keys to get access to it back.

If you want to use several wallets, you have to either run several monero-wallet-rpc instances, or load/save the wallets when appropriate. Like a file editor. If you want to edit a file and you have another file loaded, you load the other file you want to modify first.

**Comment by aaronovz1** [comment]
@moneromooo-monero any comments on why the check doesn't work?

**Comment by moneromooo-monero** [comment]
Did you test it ? You mentioned using RPC, but not the CLI wallet.

**Comment by aaronovz1** [comment]
No, I assumed RPC went through the same code path and into `simple_wallet::make_multisig`. I can test with CLI.

**Comment by moneromooo-monero** [comment]
RPC is wallet_rpc_server.cpp. Both it and simplewallet.cpp use the wallet2 library.
